engine/profiler/src/profile_render.cpp

```diff
@@ -352,26 +350,104 @@ namespace dmProfileRender
+
         char buffer[256];
 
         dmRender::DrawTextParams params;
         params.m_FaceColor   = TITLE_FACE_COLOR;
         params.m_ShadowColor = TITLE_SHADOW_COLOR;
 
-        ProfilerFrame* frame = render_profile->m_ActiveFrame ? render_profile->m_ActiveFrame : render_profile->m_CurrentFrame;
-        std::sort(frame->m_Threads.Begin(), frame->m_Threads.End(), ThreadSortTimePred());
-
-
         ProfilerThread* thread = GetSelectedThread(render_profile, frame);
 
         uint64_t ticks_per_second   = render_profile->m_TicksPerSecond;
         uint64_t max_frame_time     = render_profile->m_MaxFrameTime;
         uint64_t frame_time         = thread->m_SamplesTotalTime;
+        uint32_t drawcalls          = 0;
 
         float frame_time_f = frame_time / (float)ticks_per_second;
         float max_frame_time_f = max_frame_time / (float)ticks_per_second;
 
-        int l = dmSnPrintf(buffer, sizeof(buffer), "Frame: %6.3f Max: %6.3f", frame_time_f, max_frame_time_f);
+        {
+            // Properties
+            int y = properties_area.p.y + properties_area.s.h - LINE_SPACING;
+
+            int name_x  = properties_area.p.x;
+            int count_x = name_x + COUNTERS_NAME_WIDTH + CHARACTER_WIDTH;
+
+            params.m_FaceColor = TITLE_FACE_COLOR;
+            params.m_Text      = "Properties:";
+            params.m_WorldTransform.setElem(3, 0, name_x);
+            params.m_WorldTransform.setElem(3, 1, y);
+            dmRender::DrawText(render_context, font_map, 0, batch_key, params);
+
+            for (uint32_t i = 0; i < frame->m_Properties.Size(); ++i)
+            {
+                const ProfilerProperty& property = frame->m_Properties[i];
+
+                y -= LINE_SPACING;
+
+                if (y < (properties_area.p.y + LINE_SPACING))
+                {
+                    break;
+                }
+
+                float col[3];
+                uint32_t color_index = (property.m_NameHash >> 6) & 0x1f;
+                HslToRgb2(color_index / 31.0f, 1.0f, 0.65f, col);
+
+                if (property.m_Type == dmProfile::PROPERTY_TYPE_GROUP)
+                    params.m_FaceColor = Vector4(col[0], col[1], col[2], 1.0f);
+                else
+                    params.m_FaceColor = Vector4(1.0f);
+
+                float x = property.m_Indent * INDENT_WIDTH;
+                params.m_WorldTransform.setElem(3, 0, name_x + x);
+                params.m_WorldTransform.setElem(3, 1, y);
+
+                const char* name = dmHashReverseSafe32(property.m_NameHash);
+                if (name[0]=='r' && name[1]=='m' && name[2]=='t' && name[3]=='p' && name[4]=='_')
+                {
+                    name = name + 5;
+                }
+                params.m_Text = name;
+
+                if (name[0]=='D' && name[1]=='r' && name[2]=='a' && name[3]=='w' && name[4]=='C' && name[5]=='a' && name[6]=='l'
+                    && name[7]=='l' && name[8]=='s')
```

Comment:
ugly, but fast an simple
I think it's good enough for the profiler and a single use case